frontend/src/pages/Search/Search.js

```diff
@@ -114,42 +118,60 @@ class Search extends Component {
         const date = new Date(this.state.beginDate);
         const dateEnd = new Date(date);
         dateEnd.setHours(23, 59, 59);
+        const { start, endDest } = this.state;
         const query = {
-            originCoords: this.state.originCoords,
-            destCoords: this.state.destCoords,
+            originCoords: [start.lng, start.lat],
+            destCoords: [endDest.lng, endDest.lat],
             beginDate: date,
             endDate: dateEnd,
             distance: this.state.distance,
         };
-        const xurl = '/api/rides?' + querystring.stringify({'query': JSON.stringify(query)});
+        const xurl =
+            '/api/rides?' +
+            querystring.stringify({ query: JSON.stringify(query) });
         fetch(xurl)
-            .then(res => res.json())
-            .then(res => {
-                if (DEBUG) { console.log(res); }
-                const queried_rides = [];
-                for (let ride in res) {
-                    queried_rides.push(
-                        [
-                            res[ride].startLoc.city + ', ' + res[ride].startLoc.state,
-                            res[ride].endLoc.city + ', ' + res[ride].endLoc.state,
-                            new Date(res[ride].time),
-                        ]
-                    )
+            .then((res) => res.json())
+            .then((res) => {
+                if (DEBUG) {
+                    console.log(res);
                 }
+                // const queried_rides = [];
+                // for (let ride in res) {
+                //     queried_rides.push([
+                //         res[ride].startLoc.city +
+                //             ', ' +
+                //             res[ride].startLoc.state,
+                //         res[ride].endLoc.city + ', ' + res[ride].endLoc.state,
+                //         new Date(res[ride].time),
+                //     ]);
+                // }
+                const queried_rides = res.map((ride) => ({
```

Comment:
@alex-dinh this might break some code because previously the attributes were stored in an array but I changed it to a dictionary because that might be more scalable

Reply:
Good call using .map(), looks great